lib/tests/test_rate_map.c

```diff
@@ -177,6 +177,18 @@ test_translate_position_and_recomb_mass(void)
     /* inside recombination interval */
     CU_ASSERT_EQUAL(rate_map_position_to_mass(&map, 8), 18);
 
+    /* max valid position */
+    CU_ASSERT_EQUAL(rate_map_position_to_mass(&map, 20), 25);
+
+    /* NaNnoying cases */
+#ifdef NDEBUG // test via `meson -Db_ndebug=true ndebug_build`
+    CU_ASSERT(isnan(rate_map_position_to_mass(&map, NAN)));
```

Comment:
Would it be simpler to test the ``isfinite(value)`` and check the bounds for the parameter? Seems like an equivalent amount of computation/branching, and it would simplify the logic a bit.

Reply:
In the implementation of rate_map_position_to_mass in this commit there isn't any special handling for NaN as input. It's as simply as it would be ignoring the possibility of NaN.

One of the properties of NaN that I find rather elegant and almost beautiful is how often NaN gracefully propagates a state of erroneous computation without any special logic or handling. One of the ways I like to think of NaN is as "Can Not Be Computed/Determined". So if a "can't-be-determined" is passed in then it makes sense that "can't-be-determined" is passed right back out.
But that wasn't really the motivation for the current impl of rate_map_position_to_mass in this commit. The test is just there to make sure something like crashing doesn't happen in the absence of NDEBUG disabled asserts.